### PR TITLE
New Rule: max-file-line-count

### PR DESCRIPTION
**Pull request by chrismbarr**
Adding a new rule that limits the number of lines that a file can contain.  When files go over this limit it's usually a sign that the code is too complex, or it has too many responsibilities and it needs to be broken up into smaller pieces. 

The goal here is to count the number of lines, and it should match the line number display in your IDE _(meaning things like blank lines, comments, etc. are included in this count)_

If the rule disabling comment `/* tslint:disable:max-file-line-count */` is added anywhere in the file, this rule is ignored for the file.

### CODE REVIEW

**Comment by JKillian** [comment]
I think `file-max-line-count` would fall more in line with how other rules are named

File: src/rules/maxFileLineCountRule.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+    public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
+        const ruleFailures: Lint.RuleFailure[] = [];
+        const lineLimit: number = this.getOptions().ruleArguments[0];
+        const lineCount: number = sourceFile.getLineStarts().length;
+        const disabledIntervals = this.getOptions().disabledIntervals;
+
+        if (lineCount > lineLimit && disabledIntervals.length === 0) {
+            const errorString = Rule.FAILURE_STRING_FACTORY(lineCount, lineLimit);
+            ruleFailures.push(new Lint.RuleFailure(sourceFile, 0, 0, errorString, this.getOptions().ruleName));
```
Comment (JKillian) [inline src/rules/maxFileLineCountRule.ts]: Instead of making this a zero-length error, can we make it either one character long or the length of the first line?

File: src/tsconfig.json [hunk 2]
```diff
@@ -121,6 +122,7 @@
         "rules/oneLineRule.ts",
         "rules/oneVariablePerDeclarationRule.ts",
         "rules/onlyArrowFunctionsRule.ts",
+        "rules/orderedImportsRule.ts",
```
Comment (JKillian) [inline src/tsconfig.json]: Not sure why this change is in this diff?

File: test/rules/max-file-line-count/default/test.ts.lint [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+        /*
+        Here is another comment
+        It's very helpful.
+        */
+
+        public giveNumber = () =>{  
+            alert(this.num);
+        };
+    }
+}
```
Comment (JKillian) [inline test/rules/max-file-line-count/default/test.ts.lint]: newlines preferred at the end of files

**Comment by JKillian** [comment]
Looking good, thanks @chrismbarr! Can you make the small changes noted above and merge `master` into this?

**Comment by JKillian** [comment]
Let's just stay with `max-file-line-count` as the name, no need to change it

File: src/tsconfig.json (hunk 2 above)
Comment (chrismbarr) [inline src/tsconfig.json]: This was added when I ran the `grunt` task.  Should I just not check in this file and ignore it manually?

Comment (JKillian) [inline src/tsconfig.json]: All good, if Grunt did it it's probably right

**Comment by JKillian** [comment]
Thanks @chrismbarr, merged!
